File: src/GS.Certifications.Application/Commons/Dtos/ImpuestoDto.cs
using AutoMapper;
using GSF.Application.Common.Mappings;
using GS.Certifications.Domain.Entities.Impuestos;

namespace GS.Certifications.Application.Commons.Dtos;

public class ImpuestoDto : IMapFrom<Impuesto>
{
    public short Idm { get; set; }
    public string Nombre { get; set; }
    public string Descripcion { get; set; }
    public short? TipoId { get; set; }
    public long? ProvinciaId { get; set; }
    public decimal Alicuota { get; set; }

    public class MappingProfile : Profile
    {
        public MappingProfile()
        {
            CreateMap<Impuesto, ImpuestoDto>()
                .ForMember(dst => dst.Idm, opt => opt.MapFrom(src => src.Id))
                .ForMember(dst => dst.Alicuota, opt => opt.MapFrom(src => src.Alicuota != null ? src.Alicuota.Valor : src.Valor))
                ;
        }
    }
}

File: src/GS.Certifications.Web/wwwroot/src/Proveedores/Impuestos/ImpuestoDto.js
export default class ImpuestoDto {
    constructor(dto) {
        this.id = dto ? dto.id : null;
        this.rowVersion = dto ? dto.rowVersion : null;
        this.nombre = dto ? dto.nombre : null;
        this.descripcion = dto ? dto.descripcion : null;
        this.tipo = dto ? dto.tipo : null;
        this.provincia = dto ? dto.provincia : null;
        this.company = dto ? dto.company : null;
        this.alicuota = dto ? dto.alicuota : null;
        this.valor = dto ? dto.valor : null;
    }
}